File: leetcode/src/Num461.java
public class Num461 {
    /**
     * 思路：两个数进行异或，然后统计1的个数*/
    class Solution {
        public int hammingDistance(int x, int y) {
            int z = x ^ y;
            int res = 0;
            while(z > 0){
                res += z % 2;
                z /= 2;
            }
            return res;
        }
    }
}

File: learnPython/num461.py
# -*- coding: utf-8 -*-
"""
Created on Thu Oct 25 20:44:07 2018

@author: Fay
"""

class Solution(object):
    def hammingDistance(self, x, y):
        """
        :type x: int
        :type y: int
        :rtype: int
        """
        z = x ^ y
        res = 0
        while z > 0:
            res = res + z % 2
            z = z // 2
        return res